widget/embedded/src/pages/Home.tsx

```diff
@@ -161,53 +217,85 @@ export function Home() {
         ),
       }}>
       <Container>
-        <FromContainer>
-          <TokenInfo
-            type="From"
-            chain={fromChain}
-            token={fromToken}
-            onAmountChange={setInputAmount}
-            inputAmount={inputAmount}
-            fromChain={fromChain}
-            toChain={toChain}
-            loadingStatus={loadingMetaStatus}
-            inputUsdValue={numberToString(inputUsdValue)}
-            fromToken={fromToken}
-            setInputAmount={setInputAmount}
-            connectedWallets={connectedWallets}
-            bestRoute={bestRoute}
-            fetchingBestRoute={fetchingBestRoute}
-            onChainClick={() => navigate(navigationRoutes.fromSwap)}
-            onTokenClick={() => navigate('from-token')}
-            tokenBalanceReal={tokenBalanceReal}
-            tokenBalance={tokenBalance}
+        <InputsContainer>
+          <FromContainer>
+            <SwapInput
+              label="From"
+              onInputChange={setInputAmount}
+              balance={tokenBalance}
+              chain={{
+                displayName: fromChain?.displayName || '',
+                image: fromChain?.logo || '',
+              }}
+              token={{
+                displayName: fromToken?.symbol || '',
+                image: fromToken?.image || '',
+              }}
+              onClickToken={() => navigate('from-swap')}
+              price={{
+                value: inputAmount,
+                usdValue: numberToString(inputUsdValue),
+              }}
+              disabled={loadingMetaStatus === 'failed'}
+              loading={loadingMetaStatus === 'loading'}
+              onSelectMaxBalance={() => {
+                if (tokenBalance !== '0') {
+                  setInputAmount(tokenBalanceReal.split(',').join(''));
+                }
+              }}
+            />
+            <SwithFromAndToButton />
+          </FromContainer>
+          <SwapInput
+            label="To"
+            chain={{
+              displayName: toChain?.displayName || '',
+              image: toChain?.logo || '',
```

Comment:
In my opinion, it is better to have a default chain and token.
Because first, when the page is loaded, the image source tag is empty and it is displayed without an image.

Reply:
It should be handled in the config, allowing users to set their preferred default chain and token. However, I addressed the situation like this: if the meta is loaded but no token is selected, an image placeholder appears. The handling of its loading status is being implemented in another [PR](https://github.com/rango-exchange/rango-client/pull/342).